code/game/machinery/harvester.dm

```diff
@@ -26,9 +26,12 @@
 	. = ..()
 	interval = 0
 	var/max_time = 40
-	for(var/datum/stock_part/micro_laser/micro_laser in component_parts)
```

Comment:
You get the point. Find all occurrences and put it back

Reply:
There's nothing faulty about this approach if we are just checking for one stock part type `/datum/stock_part/micro_laser`. In other cases using if else made sense as you were checking multiple stock part types not here